drivers/place/rooms/present.cr

```diff
@@ -0,0 +1,157 @@
+require "placeos-driver"
+require "placeos-driver/interface/powerable"
+require "placeos-driver/interface/switchable"
+
+struct Enum
+  def to_json(json : JSON::Builder)
+    json.string(to_s.underscore)
+  end
+end
+
+class Place::Rooms::Present < PlaceOS::Driver
+  generic_name :System
+  descriptive_name "Presentation space logic"
+  description <<-DESC
+    Room level state and behaviours.
+
+    This driver provides a high-level API for interaction with devices, systems
+    and integrations found within common presentation spaces. It's behavior will
+    adapt to match the capabilities and configuration of other drivers present
+    in the same system.
+    DESC
+
+  accessor proj : Projector_1, implementing: InputSelection
+  accessor prev : Display_1, implementing: InputSelection
+  accessor lcd1 : Display_2, implementing: InputSelection
+  accessor lcd2 : Display_3, implementing: InputSelection
+  accessor lcd3 : Display_4, implementing: InputSelection
+  accessor switcher : Switcher_1, implementing: Switchable
+
+  def on_load
+    load_mock_state
+    on_update
+  end
+
+  enum InputType
```

Comment:
could the logic below be batched into some sort of `Routeable` interface? or is that the basis of the `Router` class